lib/server/parse-websocket.js

```diff
@@ -32,5 +33,15 @@ function parseWebSocketRequest (socket, params) {
   )
   params.compact = -1 // return full peer objects (used for websocket responses)
 
+  params.ip = opts.trustProxy
+      ? socket.upgradeReq.headers['x-forwarded-for'] || socket.upgradeReq.connection.remoteAddress
+      : (socket.upgradeReq.connection.remoteAddress && socket.upgradeReq.connection.remoteAddress.replace(common.REMOVE_IPV4_MAPPED_IPV6_RE, '')) // force ipv4
```

Comment:
Why was the extra `socket.upgradeReq.connection.remoteAddress &&` check needed here? The http case doesn't have that. Can it be removed?